pkg/batchmapper/examples/batchmap-flatmap/main.go

```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"context"
+	"log"
+	"strings"
+
+	"github.com/numaproj/numaflow-go/pkg/batchmapper"
+)
+
+func batchMapFn(_ context.Context, datums []batchmapper.Datum) batchmapper.BatchResponses {
```

Comment:
Can we do a channel input here? This will be consistent as the sinker API, and performance wise, it's slight better.

`batchMapFn(_ context.Context, datums <-chan batchmapper.Datum) batchmapper.BatchResponses {`

Reply:
@yhl25 wdyt?

Reply:
From my side I think 
1) From a efficiency stand point, I guess having a channel is better as it might help not buffer the input requests multiple times. As we first buffer on the SDK side and then pass this onto the user which would probably iterate over this as well to read the inputs. 

2) From an understanding and interface side I guess having an array is more closer to the _batch_ philosophy i guess than an iterator. Where you have a concrete list of elements in front of you and can choose to do whatever further with them.

Reply:
+1 on consistency though

Reply:
I was looking more inclined towards the array approach because of simplicity, but since we are using a channel in the sink lets go with the same here as well to be consistent.